xarray/namedarray/_array_api.py

```diff
@@ -144,3 +146,79 @@ def real(
     xp = _get_data_namespace(x)
     out = x._new(data=xp.real(x._data))
     return out
+
+
+# %% Statistical Functions
```

Comment:
@Illviljan, i'm curious to hear your thoughts on adopting a similar structure/organization as that used in Numpy so as to  it easier to incorporate new array API compliant functionality in the future. the new structure would group functionality under `/namedarray/_array_api/*.py` instead of of putting everything in `_array_api.py`.

e.g. 

```
├── _creation_functions.py
├── _data_type_functions.py
├── _dtypes.py
├── _elementwise_functions.py
├── _indexing_functions.py
├── _manipulation_functions.py
├── _searching_functions.py
├── _set_functions.py
├── _sorting_functions.py
├── _statistical_functions.py
├── linalg.py
```

Reply:
Sure! I just thought it was too few functions to bother with it right now. 

Cubed has similar structure too.

Reply:
perfect! i intend to add a few more functions to the `array_api` and i'm happy to put this structure in place